selfdrive/controls/controlsd.py

```diff
@@ -171,6 +172,11 @@ def update_events(self, CS):
     if self.sm['thermal'].memUsedPercent > 90:
       self.events.add(EventName.lowMemory)
 
+    if self.hw_type in [HwType.uno, HwType.dos]:
+      if self.sm['health'].fanSpeedRpm == 0 and self.last_desired_fan_speed > 50:
+        self.events.add(EventName.fanMalfunction)
+      self.last_desired_fan_speed = self.sm['thermal'].fanSpeed
```

Comment:
Why store last_desired fan speed? Controlsd runs at 100 Hz, so that wouldn't be enough time for the fan to spin up + the health packet being received (500 ms).

You probably want to make sure that fan speed is above 50% for at least a second.

Reply:
Ah okay, I'll store the last time the fan was seen spinning, and make sure that is over 5 seconds. I'll also switch to the current thermal fanSpeed for the desired fan speed check.